include/ola/rdm/ResponderTagSet.h

```diff
@@ -0,0 +1,221 @@
+
+#ifndef INCLUDE_OLA_RDM_RESPONDERTAGSET_H_
+#define INCLUDE_OLA_RDM_RESPONDERTAGSET_H_
+
+#include <ola/Logging.h>
+#include <ola/strings/Utils.h>
+
+#include <algorithm>
+#include <iomanip>
+#include <set>
+#include <string>
+
+namespace ola {
+namespace rdm {
+
+/**
+ * @addtogroup rdm_resp
+ * @{
+ * @class TagSet
+ * @brief Represents a set of Tags.
+ * @}
+ */
+class TagSet {
+ public:
+    /**
+     * @brief the Iterator for a TagSets
+     */
+    typedef std::set<std::string>::const_iterator Iterator;
+
+    /**
+     * @brief Construct an empty set
+     */
+    TagSet() {}
+
+    /**
+     * @brief Copy constructor.
+     * @param other the TagSet to copy.
+     */
+    TagSet(const TagSet &other):
+      m_tags(other.m_tags) {
+    }
+
+    /**
+     * @brief Assignment operator
+     */
+    TagSet& operator=(const TagSet &other) {
+      if (this != &other) {
+        m_tags = other.m_tags;
+      }
+      return *this;
+    }
+
+    /**
+     * @brief Remove all members from the set.
+     */
+    void Clear() {
+      m_tags.clear();
+    }
+
+    /**
+     * @brief Return the number of Tags in the set.
+     * @return the number of Tags in the set.
+     */
+    unsigned int Size() const {
+      return static_cast<unsigned int>(m_tags.size());
+    }
+
+    /**
+     * @brief Return whether the Tag set is empty.
+     * @return true if there are no Tags in the set.
+     */
+    bool Empty() const {
+      return m_tags.empty();
+    }
+
+    /**
+     * @brief Add a Tag to the set.
+     * @param tag the Tag to add.
+     */
+    void AddTag(const std::string &tag) {
+      m_tags.insert(tag);
+    }
+
+    /**
+     * @brief Remove a Tag from the set.
+     * @param tag the Tag to remove.
+     */
+    void RemoveTag(const std::string &tag) {
+      m_tags.erase(tag);
+    }
+
+    /**
+     * @brief Check if the set contains a Tag.
+     * @param tag the Tag to check for.
+     * @return true if the set contains this Tag.
+     */
+    bool Contains(const std::string &tag) const {
+      return m_tags.find(tag) != m_tags.end();
+    }
+
+    /**
+     * @brief Return an Iterator to the first member of the set.
+     */
+    Iterator Begin() const {
+      return m_tags.begin();
+    }
+
+    /**
+     * @brief Return an Iterator to one-past-the-last member of the set.
+     */
+    Iterator End() const {
+      return m_tags.end();
+    }
+
+    /**
+     * @brief Equality operator.
+     * @param other the TagSet to compare to.
+     */
+    bool operator==(const TagSet &other) const {
+      return m_tags == other.m_tags;
+    }
+
+    /**
+     * @brief Inequality operator.
+     * @param other the TagSet to compare to.
+     */
+    bool operator!=(const TagSet &other) const {
+      return !(*this == other);
+    }
+
+    /**
+     * @brief Convert a TagSet to a human readable string.
+     * @returns a comma separated string with the Tags from the set.
+     * @note Commas can also exist within the tags themselves...
```

Comment:
Thoughts on how best to handle this? I don't think any of our other ToStrings return a multi-line string which would be the other obvious way to list them. Also probably not too handy in general usage, so I guess just risking the confusing comma is probably worth it.

Reply:
I'd missed that you'd approved it as well as the comments. I'll merge this ready for the next one anyway @kripton . But would be curious on a second point of view on this point...